fix(curriculum): updated tests to fix attribute sensitivity order (#52813)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/65672136535209761a5cf02b.md

# --hints--

> You do not need to change the other parts of the code. All you need to do is add the `onclick` attribute to the button element.

Reviewer: I'm not crazy about this sentence. I think if we fix the tests so the order doesn't matter we can just simplify the original sentence.

```suggestion
You should add an `onclick` attribute with the value `playSong(${song.id})` to the `button` element.
```

Reviewer: I think all we care about here is that the handler was added correctly. Then we can test that the button element still has the class on it in the next test.

```js

> assert.match(code, /<button\s.*>\s*<span\s+class=('|")playlist-song-title\1>\$\{song\.title\}<\/span>\s*<span\s+class=\1playlist-song-artist\1>\$\{song\.artist\}<\/span>\s*<span\s+class=\1playlist-song-duration\1>\$\{song\.duration\}<\/span>\s*<\/button>/)

Reviewer 2: @jdwilkin4 Are we not able to test the document object? E.g. `assert.includes(document.querySelector("button").classList, "playlist-song-title")`.

Reviewer 3: > Are we not able to test the document object?

I wasn't aware there was an issue with that. That should work

Reviewer: Testing that everything remains the same seems out of scope for this step. If we had to do that every time our tests would become bloated.

The test message also becomes unclear, "something that shouldn't change, changed" isn't a very helpful message. The regex is just too large for a single assert message and it won't provide useful feedback.

I think all we do here is check that the class remains on the element after the handler is added.

```

You do not need to change the `class` attribute. All you need to do is add the `onclick` attribute to the button element.

```js
assert.match(code, /\s*class=('|")playlist-song-info\1\s*/)

> ```

Reviewer: Not sure we need this.
